modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/policy/CompositeArtifactExtensionManager.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.runtime.module.deployment.impl.internal.policy;
+
+import static java.lang.String.format;
+import static org.mule.runtime.api.util.Preconditions.checkArgument;
+import org.mule.runtime.api.meta.model.ExtensionModel;
+import org.mule.runtime.core.api.Event;
+import org.mule.runtime.core.api.extension.ExtensionManager;
+import org.mule.runtime.extension.api.manifest.ExtensionManifest;
+import org.mule.runtime.extension.api.runtime.ConfigurationInstance;
+import org.mule.runtime.extension.api.runtime.ConfigurationProvider;
+
+import java.net.URL;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Composes {@link ExtensionManager} from a child and a parent artifacts, so child artifact can access extensions provided by the
+ * parent.
+ */
+public class CompositeArtifactExtensionManager implements ExtensionManager {
+
+  private final ExtensionManager parentExtensionManager;
+  private final ExtensionManager childExtensionManager;
```

Comment:
Understood, there was an error on my previous sentence. What I meant was that mule artifact can have only a parent, an because this composite is for a child artifact, is a 1-1 relation. One parent, one child. No need for a list. Unless I missing something